---
version: '3.6'

services:
  zookeeper:
    image: wurstmeister/zookeeper:3.4.6
    ports:
      - "2181:2181"

  kafka:
    image: wurstmeister/kafka:2.13-2.8.1
    ports:
      - "9091:9092"
    expose:
      - 9091
    environment:
      KAFKA_ADVERTISED_LISTENERS: INSIDE://kafka:9091,OUTSIDE://localhost:9092
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
      KAFKA_LISTENERS: INSIDE://0.0.0.0:9091,OUTSIDE://0.0.0.0:9092
      KAFKA_INTER_BROKER_LISTENER_NAME: INSIDE
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  kafka-consumer:
    container_name: kafka_consumer
    build:
      context: ./consumer
    image: ppgti/kafka-consumer
    command: python app.py
    ports:
      - "5001:5001"
    volumes:
      - ./consumer/:/app
#    restart: on-failure
    depends_on:
      - kafka
    links:
      - kafka

  kafka-producer:
    container_name: kafka_producer
    build:
      context: ./producer
    image: ppgti/kafka-producer
    command: python app.py
    ports:
      - "5002:5002"
    volumes:
      - ./producer/:/app
    links:
      - kafka
